worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -1478,29 +1482,33 @@ public int makeSphere(Vector pos, Pattern block, double radiusX, double radiusY,
                         }
                     }
 
-                    if (setBlock(pos.add(x, y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(-x, y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(x, -y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(x, y, -z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(-x, -y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(x, -y, -z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(-x, y, -z), block)) {
-                        ++affected;
+                    if(!hemi || (hemi && !upsideDown)) {
```

Comment:
For readability, I would just put two booleans at the top, `makeTop` and `makeBottom`, and set them to whatever expression. Then use those in the `if` statements.